src/main/java/org/jboss/tools/intellij/openshift/actions/cluster/LoginAction.java

```diff
@@ -37,18 +37,18 @@ public LoginAction() {
   public String getTelemetryActionName() {return "login to cluster";}
 
   @Override
-  public void actionPerformedOnSelectedObject(AnActionEvent anActionEvent, Object selected, @NotNull OdoFacade odo) {
+  public void actionPerformedOnSelectedObject(AnActionEvent anActionEvent, Object selected, @NotNull Oc oc) {
```

Comment:
I found a problem with the `LoginDialog` but I dont know where the problem is exactly:

**Steps:**
1. ASSERT: make sure that you dont have oc in the cache:
`rm -rf ~/.oc/cache`
2. EXEC: pick "Log in to Cluster" in the ctx menu of the cluster node
3. ASSERT: you see "Downloading oc" with a progress bar in the bottom right corner of the IDE window

**Result:**
Nothing happens, the Login dialog does not show up. You have to pick again "Log in to Cluster" in the context menu.

Reply:
can reproduce this. investigating

Reply:
ok, debugged and turns out the first call on login returns a null OcTool as completable future is not complete, so the login doesn't shows up. in the background the tool is downloaded, then in the second call, the tool is correclty returned. will modify the logic.